pgjdbc/src/main/java/com/yugabyte/ysql/TopologyAwareLoadBalancer.java

```diff
@@ -134,6 +134,68 @@ protected void updateCurrentHostList(ArrayList<String> currentPrivateIps, String
     }
   }
 
+  @Override
+  public int getPriority(String cloud,String region, String zone){
+    CloudPlacement cp = new CloudPlacement(cloud, region, zone);
+    int priotrity = getKeysByValue(allowedPlacements,cp);
+    return priotrity;
+  }
+
+  public int getKeysByValue(Map<Integer, Set<CloudPlacement>> allowedPlacements, CloudPlacement cp) {
```

Comment:
Can be a `private` method and `allowedPlacements` does not need to passed as method parameter. It is available as member variable.
```
private int getKeysByValue(CloudPlacement cp)
```